openeo-geotrellis/src/main/scala/org/openeo/geotrellis/ComputeStatsGeotrellisAdapter.scala

```diff
@@ -150,96 +236,17 @@ class ComputeStatsGeotrellisAdapter(zookeepers: String, accumuloInstanceName: St
     finally
       statisticsWriter.close()
   }
 
-  private def projectedPolygons(vector_file: String): (Array[MultiPolygon], CRS) = {
-    val vectorUrl = try {
-      new URL(vector_file)
-    } catch {
-      case _: MalformedURLException => new URL(s"file://$vector_file")
-    }
-
-    val filename = vectorUrl.getPath.split("/").last
-
-    if (filename.endsWith(".shp")) readSimpleFeatures(vectorUrl)
-    else readMultiPolygonsFromGeoJson(vectorUrl)
-  }
-
-  // adapted from Geotrellis' ShapeFileReader to avoid having too much in memory
-  private def readSimpleFeatures(shpUrl: URL): (Array[MultiPolygon], CRS) = {
-    val ds = new ShapefileDataStore(shpUrl)
-    val ftItr: SimpleFeatureIterator = ds.getFeatureSource.getFeatures.features
-
-    try {
-      val featureCount = ds.getCount(Query.ALL)
-      require(featureCount < Int.MaxValue)
-
-      val simpleFeatures = new Array[MultiPolygon](featureCount.toInt)
-
-      for (i <- simpleFeatures.indices) {
-        val multiPolygon = ftItr.next().getAttribute(0) match {
-          case multiPolygon: MultiPolygon => multiPolygon
-          case polygon: Polygon => MultiPolygon(polygon)
-        }
-
-        simpleFeatures(i) = multiPolygon
-      }
-
-      // FIXME: read it from the shp and default to LatLng
-      (simpleFeatures, LatLng)
-    } finally {
-      ftItr.close()
-      ds.dispose()
-    }
-  }
-
-  private def readMultiPolygonsFromGeoJson(geoJsonUrl: URL): (Array[MultiPolygon], CRS) = {
-    // FIXME: stream it instead
-    val src = Source.fromURL(geoJsonUrl)
-
-    val multiPolygons = try {
-      val geoJson = src.mkString
-
-      def children(geometryCollection: GeometryCollection): Stream[Geometry] = {
-        def from(i: Int): Stream[Geometry] =
-          if (i >= geometryCollection.getNumGeometries) Stream.empty
-          else geometryCollection.getGeometryN(i) #:: from(i + 1)
-
-        from(0)
-      }
-
-      def asMultiPolygons(geometry: Geometry): Array[MultiPolygon] = geometry match {
-        case polygon: Polygon => Array(MultiPolygon(polygon))
-        case multiPolygon: MultiPolygon => Array(multiPolygon)
-        case geometryCollection: GeometryCollection => children(geometryCollection).map {
-          case polygon: Polygon => MultiPolygon(polygon)
-          case multiPolygon: MultiPolygon => multiPolygon
-        }.toArray
-      }
-
-      try {
-        asMultiPolygons(geoJson.parseGeoJson[Geometry]())
-      } catch {
-        case _: DecodingFailure =>
-          val featureCollection = geoJson.parseGeoJson[JsonFeatureCollection]()
-          featureCollection.getAllGeometries()
-            .flatMap(asMultiPolygons)
-            .toArray
-      }
-    } finally src.close()
-
-    (multiPolygons, LatLng)
-  }
-
   def compute_histogram_time_series(product_id: String, polygon_wkt: String, polygon_srs: String,
```

Comment:
Since `ProjectedPolygons` becomes the primary input abstraction of these methods, we might want to reduce the surface area of the interface and get rid of methods that accept `polygon_wkt(s)` and `polygon(s)_srs`; unit tests can be adapted by calling the ProjectedPolygons primary constructor.

`compute_histogram_time_series` and `compute_histogram_time_series_from_datacube` look like they can be removed, eliminating the need to expose `ProjectedPolygons#parsePolygonWkt` as public: you can make it private instead.

Reply:
indeed, I was thinking the same, but was not sure if I could just drop existing methods